lib/Echidna/Solidity.hs

```diff
@@ -106,8 +112,11 @@ contracts fp = let usual = ["--solc-disable-warnings", "--export-format", "solc"
                   (\sa -> if null sa then [] else ["--solc-args", sa])
   maybe (throwM CompileFailure) (pure . toList . fst) =<< liftIO (do
     stderr <- if q then UseHandle <$> openFile "/dev/null" WriteMode else pure Inherit
-    _ <- readCreateProcess (proc "crytic-compile" $ (c ++ solargs) |> fp) {std_err = stderr} ""
-    readSolc "crytic-export/combined_solc.json")
+    (ec, _, _) <- readCreateProcessWithExitCode (proc "crytic-compile" $ (c ++ solargs) |> fp) {std_err = stderr} ""
```

Comment:
We can use the full path returned by `findExecutable`

Reply:
I am in favor of this

Reply:
We could use the path returned from the `findExecutable` call, however, if it finds path, then `readCreateProcessWithExitCode` should not fail (with code 127).

Reply:
It still might fail due to a race

Reply:
Fair enough. This should be implemented already. Please take a look.